test/03_device-state.ts

```diff
@@ -1799,6 +1799,108 @@ export default () => {
 						});
 					});
 				}
+
+				describe('Reporting no running release', () => {
+					beforeEach(async function () {
+						await fakeDevice.patchState(
+							device,
+							device.uuid,
+							{
+								[stateKey]:
+									stateVersion === 'v2'
+										? {
+												is_on__commit: release2.commit,
+											}
+										: {
+												apps: {
+													[applicationUuid]: {
+														release_uuid: release2.commit,
+													},
+												},
+											},
+							},
+							stateVersion,
+						);
+
+						await expectResourceToMatch(pineUser, 'device', device.id, {
+							is_running__release: { __id: release2.id },
+						});
+					});
+				});
+
+				it('should clear the current release when the device reports a null commit', async () => {
+					await fakeDevice.patchState(
+						device,
+						device.uuid,
+						{
+							[stateKey]:
+								stateVersion === 'v2'
+									? {
+											is_on__commit: null,
+										}
+									: {
+											apps: {
+												[applicationUuid]: {
+													release_uuid: null,
```

Comment:
Not sure this would ever happen. The `release_uuid` property would just be undefined

Reply:
I realized, but I wanted to make sure that in case we explicitly pass `null` in this prop, then it is indeed getting stored, b/c that's what makes sense to do any way.
Effectively I treat the supervisor omitting the `release_uuid` when it's `null`, just as an optimization, and make sure that the basic semantics work as well (even though we don't  expect it to happen).